nomos-services/http/src/http.rs

```diff
@@ -188,25 +188,44 @@ pub async fn handle_graphql_req<M, F>(
     payload: Option<Bytes>,
     relay: OutboundRelay<M>,
     f: F,
-) -> Result<String, overwatch_rs::DynError>
+) -> Result<String, ServiceError>
 where
     F: FnOnce(
         async_graphql::Request,
         oneshot::Sender<async_graphql::Response>,
-    ) -> Result<M, overwatch_rs::DynError>,
+    ) -> Result<M, ServiceError>,
 {
-    let payload = payload.ok_or("empty payload")?;
+    #[derive(Debug)]
+    struct EmptyPayload;
+
+    impl core::fmt::Display for EmptyPayload {
+        fn fmt(&self, f: &mut core::fmt::Formatter<'_>) -> core::fmt::Result {
+            write!(f, "empty payload")
+        }
+    }
+
+    impl Error for EmptyPayload {}
```

Comment:
I do not think this impls should be inside the function itself. Any reason why?

Reply:
Yeah, at that moment, we do not have `Custom` for `ServiceError`. Now I add a `ServiceError::custom` fn to solve this.